darts/explainability/explainability_result.py

```diff
@@ -81,10 +103,80 @@ def get_explanation(
             ),
         )
 
-        if isinstance(self.explained_forecasts, list):
+
+class ShapExplainabilityResult(ExplainabilityResult):
+    """
+    Stores the explainability results of a :class:`ShapExplainer`
+    with convenient access to the results. It extends the :class:`ExplainabilityResult` and carries additional
+    information specific to the Shap explainers. In particular, in addition to the `explained_forecasts` (which in
+    the case of the :class:`ShapExplainer` are the shap values), it also provides access to the corresponding
+    `feature_values` and the underlying `shap.Explanation` object.
+    """
+
+    def __init__(
+        self,
+        explained_forecasts: Union[
+            Dict[integer, Dict[str, TimeSeries]],
+            Sequence[Dict[integer, Dict[str, TimeSeries]]],
+        ],
+        feature_values: Union[
+            Dict[integer, Dict[str, TimeSeries]],
+            Sequence[Dict[integer, Dict[str, TimeSeries]]],
+        ],
+        shap_explanation_object: Optional[
+            Union[
+                Dict[integer, Dict[str, shap.Explanation]],
+                Sequence[Dict[integer, Dict[str, shap.Explanation]]],
+            ]
+        ],
+    ):
+        super().__init__(explained_forecasts)
+        self.feature_values = feature_values
+        self.shap_explanation_object = shap_explanation_object
+
+    def get_feature_values(
```

Comment:
We could further reduce the code here by adding a private method to ExplainabilityResult (see below):

```
    def _query_explainability_result(
        self,
        attr: Union[Dict[integer, Dict[str, TimeSeries]], Sequence[Dict[integer, Dict[str, TimeSeries]]]],
        horizon: int,
        component: Optional[str] = None,
    ) -> Any:
        """
        Returns one or several `TimeSeries` representing the feature values
        for a given horizon and component.

        Parameters
        ----------
        attr
            An explainability result attribute from which to extract the horizon and comonent.
        horizon
            The horizon for which to return the feature values.
        component
            The component for which to return the feature values. Does not
            need to be specified for univariate series.
        """
        self._validate_input_for_querying_explainability_result(horizon, component)
        if isinstance(attr, list):
            return [attr[i][horizon][component] for i in range(len(attr))]
        else:
            return attr[horizon][component]
```

Now we can simply call this method from all the getters (explanation, features, shap expl object):

```
def get_feature_values(...):
    return self._query_explainability_result(self.feature_values, horizon=horizon, component=component)